File: serv_req.py
#Import necessary libraries
from docx import Document #docx manipulation library
from sys import argv
import os.path
from datetime import datetime

import boto3 #AWS Python SDK library
from email.mime.text import MIMEText
from email.mime.application import MIMEApplication
from email.mime.multipart import MIMEMultipart
ses = boto3.client('ses')


script, req_date,req_time,req_client,req_department,req_address,req_phone,req_contact,req_system,type,req_desc,req_nameof,engineer = argv

template = Document('S_R_F.docx')

newFile = template
tab = newFile.tables[0]


tab.cell(0,1).paragraphs[0].clear()
tab.cell(0,1).paragraphs[0].text = req_date

tab.cell(1,1).paragraphs[0].clear()
tab.cell(1,1).paragraphs[0].text = req_time

tab.cell(2,1).paragraphs[0].clear()
tab.cell(2,1).paragraphs[0].text = req_client

tab.cell(3,1).paragraphs[0].clear()
tab.cell(3,1).paragraphs[0].text = req_department

tab.cell(4,1).paragraphs[0].clear()
tab.cell(4,1).paragraphs[0].text = req_address

tab.cell(5,1).paragraphs[0].clear()
tab.cell(5,1).paragraphs[0].text = req_phone

tab.cell(6,1).paragraphs[0].clear()
tab.cell(6,1).paragraphs[0].text = req_contact

tab.cell(7,1).paragraphs[0].clear()
tab.cell(7,1).paragraphs[0].text = req_system

tab.cell(8,1).paragraphs[0].clear()
tab.cell(8,1).paragraphs[0].text = type

tab.cell(9,1).paragraphs[0].clear()
tab.cell(9,1).paragraphs[0].text = req_desc

newFileName = req_client+'_'+datetime.now().strftime("%Y-%m-%d %H:%M:%S")
newFileName = 'S_Req_'+newFileName.replace(" ", "_")+'.docx'
newFile.save(newFileName)
#newFile.close()

"""
This section is for sending the email using AWS raw email via SES
"""

message = MIMEMultipart()
message['Subject'] = 'Incoming Service Request'
message['From'] = 'CCE-Service <service@cce-service.ca>'
message['To'] = ', '.join(['sbrukson@gmail.com', 'alex@ccemedical.com'])
# message body
body_string = 'Submitted By: '+req_nameof+'<br/> Requested Engineer: '+engineer+' <br/><br/>'
body_string += 'Date: '+req_date+'<br/> Client: '+req_client+' <br/> Address: '+req_address
body_string += '<br/> Systems: '+req_system+'<br/> Description: '+req_desc


part = MIMEText(body_string, 'html')
message.attach(part)
# attachment

part = MIMEApplication(open(newFileName, 'rb').read())

part.add_header('Content-Disposition', 'attachment', filename=newFileName)
message.attach(part)
response = ses.send_raw_email(
    Source=message['From'],
    Destinations=['sbrukson@gmail.com', 'alex@ccemedical.com'],
    RawMessage={
        'Data': message.as_string()
    }
)

print(response)

File: serv_req.php
<?php
//
$form_data = array($_POST["req_date"],
	$_POST["req_time"],
	$_POST["req_client"],
	$_POST["req_department"],
	$_POST["req_address"],
	$_POST["req_phone"],
	$_POST["req_contact"],
	$_POST["req_systems"],
	$_POST["type"],
	$_POST["req_desc"]);
$service_person = $_POST["engineer"];
$bad_char = array ('\"','$');
$comm_txt = "";
for ($x =0; $x <count($form_data); $x++){
	$form_data[$x] = str_replace('"','',$form_data[$x]);
	$comm_txt = $comm_txt." \"".$form_data[$x]."\"";
}
$comm_txt."<br/>";
$command = escapeshellcmd("python serv_req.py{$comm_txt}");



$output = shell_exec($command." 2> error.log");
$output = trim($output);
echo ($output."<br/>");
?>

<?php
$name  ="CCEService";
$email  ="service@cce-service.ca";
$fetch_message  ="Service Request Incoming!<br/><br/>";
$fetch_message .= "<b>Date: </b>".$_POST["req_date"]." @ ".$_POST["req_time"]."<br/>";
$fetch_message .= "<b>Client: </b>".$_POST["req_client"]."<br/>";
$fetch_message .= "<b>Address: </b>".$_POST["req_address"]."<br/>";
$fetch_message .= "<b>Systems: </b>".$_POST["req_systems"]."<br/>";
$fetch_message .= "<b>Description: </b>".$_POST["req_desc"]."<br/>";



$fetch_subject ="Service Request";


$file_ext=explode(".",$output);
$file_type=strtolower($file_ext[1]);

$name_of_attachment = basename($output);
$tmp_path = $output;

if($file_type == "pdf" )
{
 $mime_type = "application/pdf";
}
else if($file_type == "doc" )
{
 $mime_type = "application/msword";
}
else if($file_type == "docx" )
{
 $mime_type = "application/vnd.openxmlformats-officedocument.wordprocessingml.document";
}
else {echo "None of the above my friend</br>";}

if ($file_type == 'pdf' || $file_type == 'doc' || $file_type == 'docx' )
  {
   $to = "sbrukson@gmail.com, alex@ccemedical.com";
   $subject =  $fetch_subject;
   $bound_text = "_____________________";
   $bound = "--".$bound_text."\r\n";
   $bound_last = "--".$bound_text."--\r\n";

  $headers .= "Reply-To: Service <service@cce-service.ca>\r\n";
  $headers .= "Return-Path: Service <service@cce-service.ca>\r\n";

   $headers = "From: \"".$name."\" <{$email}>\r\n";
   $headers .= "MIME-Version: 1.0\r\n"
   ."Content-Type: multipart/mixed; boundary=\"$bound_text\"";

   $message .= "".$bound;

   $message .= "Content-Type: text/html; charset=\"iso-8859-1\"\r\n"
   ."Content-Transfer-Encoding: 7bit\r\n\r\n"
   ."<b>Message: </b> ".$fetch_message." \r\n"
   .$bound;

   $file = file_get_contents($tmp_path);
   $file_encode = chunk_split(base64_encode($file));

   $message .= "Content-Type: ".$mime_type."; name=\"".$name_of_attachment."\"\r\n"
   ."Content-Transfer-Encoding: base64\r\n"
   ."Content-disposition: attachment; file=".$name_of_attachment."\r\n"
   ."\r\n"
   .$file_encode
   .$bound_last;

   if(mail($to, $subject, $message, $headers))
   {
    echo '<script language="javascript">alert("Message sent.");</script>';
    echo "<script>  location.replace(\"https://cce-service.ca\")</script>";
}
   else
   {
	echo '<script language="javascript">alert("Error in sending message. Please try again later.");</script>';
   }

  }
  else
  {
   echo 'Invalid uploaded file type!';
  }

?>
